core/chaincode/shim/crypto/ac/abac.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright IBM Corp. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		 http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ac
+
+import (
+	"bytes"
+	"crypto/x509"
+	"errors"
+
+	"github.com/hyperledger/fabric/core/crypto/abac"
```

Comment:
Agree with @muralisrini's comments, the tight coupling will make it very difficult to implement chaincode shims in other languages and is making the binaries too big.